### PR TITLE
#153 #155 Bug fixes

### PR DESCRIPTION
**Pull request by CarlosNZ**
2 small fixes -- #153 & #155

**NOTE**: If it's not already merged, this should go with back-end PR: https://github.com/openmsupply/application-manager-server/pull/130

#153 was due to the old "0 === false" trap in javascript. Needed to change to an explicit `defaultIndex !== undefined` check, because "defaultIndex" can often be 0, which would return false. :(

#155 There's probably a more thorough fix involving making sure `validationState` is properly calculated before rendering the child plugin. However, for now, having a fall-back (`isValid: true`) options will do the trick.

### CODE REVIEW

File: src/formElementPlugins/ApplicationViewWrapper.tsx [hunk 1]
```diff
@@ -81,7 +81,7 @@ const ApplicationViewWrapper: React.FC<ApplicationViewWrapperProps> = (props) =>
           <ApplicationView
             onUpdate={onUpdate}
             onSave={onSave}
-            validationState={validationState}
+            validationState={validationState || { isValid: true }}
```
Comment (nmadruga) [inline src/formElementPlugins/ApplicationViewWrapper.tsx:84]: Maybe a good idea to add a comment here to explain. From my understanding there is still a problem happening, we just don't want to get much involved with the solution for now.

**Review by nmadruga — COMMENTED** [review]
Just a comment, I don't have the details of what is causing the problem, but it might be good to describe here as a comment - so we know what to fix in the near future.

File: src/formElementPlugins/ApplicationViewWrapper.tsx (hunk 1 above)
Comment (CarlosNZ) [reply src/formElementPlugins/ApplicationViewWrapper.tsx:84]: Yeah okay. Done now in 4e35cdf

```
// TO-DO: ensure validationState gets calculated BEFORE rendering this child, so we don't need this fallback.
```

**Review by nmadruga — APPROVED** [review]
Thanks for adding the comment. Approved now

**Pull request merged**